### PR TITLE
[2.2] Shapes and SC

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
Hi,

Some fixes to FlexHCD, Shapes (for MXCuBE3) and ISPyBRestClient

- Improved the sample loading and unloading routines of FlexHCD
- Hiding grids that get out of a certain, configurable,  threshold angle automatically when omega rotates
- Handling workfloStepId's in get_dc of ISPyBRestClient
- And some other minor fixes like typos

Marcus

### CODE REVIEW

File: HardwareObjects/Shapes.py [hunk 1]
```diff
@@ -98,6 +102,7 @@ def add_shape(self, shape):
         :type shape: Shape object.
         """
         self.shapes[shape.id] = shape
+        shape.shapes_hw_object = self
```
Comment (meguiraun) [inline HardwareObjects/Shapes.py]: why adding this? I do not understand

File: HardwareObjects/Shapes.py [hunk 2]
```diff
@@ -414,11 +428,22 @@ def __init__(self, mpos_list, screen_coord):
         self.beam_pos = [1, 1]
         self.beam_width = 0
         self.beam_height = 0
+        self.hide_threshold = 5
 
         self.set_id(Grid.SHAPE_COUNT)
 
+    def update_position(self, transform):
+        phi_pos = self.shapes_hw_object.diffractometer.phiMotor.getPosition() % 360
```
Comment (meguiraun) [inline HardwareObjects/Shapes.py]: and `self.diffractometer.phiMotor....` instead?

Comment (meguiraun) [reply HardwareObjects/Shapes.py]: sorry, yes... messed up with the diff. Ignore it

**Comment by meguiraun** [comment]
sorry it took so long for review.

File: HardwareObjects/Shapes.py (hunk 1 above)
Comment (marcus-oscarsson) [reply HardwareObjects/Shapes.py]: So that the shape can access the shapes object that is "managing" it,

File: HardwareObjects/Shapes.py (hunk 2 above)
Comment (marcus-oscarsson) [reply HardwareObjects/Shapes.py]: So, I can ignore this comment then ?

Comment (meguiraun) [reply HardwareObjects/Shapes.py]: yes, just merged!

**Pull request merged**